src/components/MenuPanel/MenuPanel.js

```diff
@@ -1,119 +1,66 @@
 // @flow
 
 import classNames from 'classnames'
-import React, { Component } from 'react'
+import React, { PureComponent } from 'react'
 
-import JLogo from 'components/base/JLogo'
-import getWallet from 'utils/wallets/getWallet'
-
-import MenuPanelActions from './Actions'
-import MenuPanelMainAction from './MainAction'
-import MenuPanelBalanceTicker from './BalanceTicker'
-import MenuPanelWalletManager from './WalletManager'
+import menuPanelStyle from './menuPanel.m.scss'
+import { Top } from './Top'
+import { Back } from './Back'
+import { Actions } from './Actions'
+import { Settings } from './Settings'
+import { Separator } from './Separator'
 
 type Props = {|
-  +getMoreAddresses: () => void,
-  +setActiveAddress: (Index) => void,
-  +items: Wallets,
-  +addresses: Address[],
-  +addressNames: AddressNames,
-  +activeWalletId: ?WalletId,
-  +fiatCurrency: FiatCurrency,
+  +walletName: string,
+  +fiatCurrency: string,
+  +mnemonicAddressName: string,
+  +previousRouteNameFallback: ?string,
   +fiatBalance: number,
+  +isMnemonic: boolean,
+  +isMinimized: boolean,
 |}
 
-type ComponentState = {|
-  isActionsMoreActive: boolean,
-  isWalletManagerActive: boolean,
-|}
-
-class MenuPanel extends Component<Props, ComponentState> {
-  constructor(props: Props) {
-    super(props)
-
-    this.state = {
-      isActionsMoreActive: false,
-      isWalletManagerActive: false,
-    }
-  }
-
-  toggleActionsMore = () => {
-    this.setState({
-      isWalletManagerActive: false,
-      isActionsMoreActive: !this.state.isActionsMoreActive,
-    })
-  }
-
-  toggleWalletManager = () => {
-    this.setState({
-      isActionsMoreActive: false,
-      isWalletManagerActive: !this.state.isWalletManagerActive,
-    })
+export class MenuPanel extends PureComponent<Props> {
```

Comment:
I suggest we put `MenuPanel` under `MenuLayout`, because it is the one and only place we would use it.